src/D2L.Bmx/OktaAuthenticator.cs

```diff
@@ -130,6 +143,121 @@ private bool TryAuthenticateFromCache(
 		return true;
 	}
 
+	private async Task<IOktaAuthenticatedClient?> TryAuthenticateWithDSSOAsync(
+		string org,
+		string user,
+		IOktaClientFactory oktaClientFactory,
+		bool experimental
+	) {
+		await using IBrowser? browser = await Browser.LaunchBrowserAsync( experimental );
+		if( browser is null ) {
+			return null;
+		}
+
+		Console.WriteLine( "Attempting to automatically login using DSSO." );
+		var cancellationTokenSource = new CancellationTokenSource( TimeSpan.FromSeconds( 15 ) );
+		var sessionIdTaskProducer = new TaskCompletionSource<string?>( TaskCreationOptions.RunContinuationsAsynchronously );
+		var userEmailTaskProducer = new TaskCompletionSource<string?>( TaskCreationOptions.RunContinuationsAsynchronously );
+		string? sessionId;
+		string? userEmail;
+
+		try {
+			var page = await browser.NewPageAsync().WaitAsync( cancellationTokenSource.Token );
+			string baseAddress = $"https://{org}.okta.com/";
+			int attempt = 1;
+
+			page.Load += ( _, _ ) => _ = GetSessionCookieAsync();
+			page.Response += ( _, responseCreatedEventArgs ) => _ = GetOktaUserEmailAsync(
+				responseCreatedEventArgs.Response
+			);
+			await page.GoToAsync( baseAddress );
+			sessionId = await sessionIdTaskProducer.Task.WaitAsync( cancellationTokenSource.Token );
+			userEmail = await userEmailTaskProducer.Task.WaitAsync( cancellationTokenSource.Token );
+
+			async Task GetSessionCookieAsync() {
+				var url = new Uri( page.Url );
+				if( url.Host == $"{org}.okta.com" ) {
+					string title = await page.GetTitleAsync();
+					if( title.Contains( "sign in", StringComparison.OrdinalIgnoreCase ) ) {
+						if( attempt < 3 ) {
+							attempt++;
+							await page.GoToAsync( baseAddress );
+						} else {
+							sessionIdTaskProducer.SetResult( null );
+						}
+						return;
+					}
+				}
+				var cookies = await page.GetCookiesAsync( baseAddress );
+				if( Array.Find( cookies, c => c.Name == "sid" )?.Value is string sid ) {
+					sessionIdTaskProducer.SetResult( sid );
+				}
+			}
+
+			async Task GetOktaUserEmailAsync(
+				IResponse response
+			) {
+				if( response.Url.Contains( $"{baseAddress}enduser/api/v1/home" ) ) {
+					string content = await response.TextAsync();
+					var home = JsonSerializer.Deserialize( content, JsonCamelCaseContext.Default.OktaHomeResponse );
+					if( home is not null ) {
+						userEmailTaskProducer.SetResult( home.Login );
+					}
+				}
+			}
+
+		} catch( TaskCanceledException ) {
+			consoleWriter.WriteWarning(
+				$"WARNING: Timed out when trying to create {org} Okta session through DSSO. Check if org is correct."
+					+ " If running BMX with admin privileges, rerun the command with the '--experimental' flag."
```

Comment:
This maybe also needs some mention about being on the vpn. I don't know how to make that org agnostic though.

Maybe `If your Okta org requires a VPN connection, ensure it's active`? There's already a lot of text here